lib/m2ts/caption-stream.js

```diff
@@ -406,10 +411,45 @@ Cea708Service.prototype.setCurrentWindow = function(windowNum) {
   this.currentWindow = this.windows[windowNum];
 };
 
-var Cea708Stream = function() {
+/**
+ * Try to create a TextDecoder if it is natively supported
+ */
+Cea708Service.prototype.createTextDecoder = function(encoding) {
+  if (typeof TextDecoder === 'undefined') {
+    this.trigger('log', {
+      level: 'warn',
+      message: 'The `encoding` option is unsupported without TextDecoder support'
+    });
+  } else {
+    try {
+      this.textDecoder_ = new TextDecoder(encoding);
```

Comment:
This appears to throw if the encoding was incorrect. We may want to concat the caught error onto our warning below.